### DISCUSSION TITLE
Powershell SqlConnection Connection Timeout not honored in Powershell 7.x

### DISCUSSION THREAD

**Discussion by IanElsinga** [PowerShell/PowerShell#19832] [General]
Hi, I have spent much time looking for information on this and have found nothing. If this is not the right place to discuss this, could someone direct me to the correct place? 

With Powershell 7.x, when using the SqlServer powershell module, or using the SMO SqlConnection object directly, the ConnectionTimeout parameter is not honored (either with Invoke-SqlCmd ConnectionTimeout parameter or with "Connection Timeout=x" in the connection string).  Specifying anything below 15s results in a 15s timeout - which I assume is the default. 

But when powershell 5.1 is used, the Connection Timeout parameter IS honored, and it times out precisely according to the value of the parameter. 

Is this a bug? Is there a workaround? Many thanks for any information.

**Comment by kasini3000** [comment]
$sqlConn = New-Object System.Data.SqlClient.SqlConnection("Data Source=DataSource;Initial Catalog=DataCatalog;Integrated Security=False;Connection Timeout=10;User ID=user;Password=pwd;")

my share powershell sqlclient script ：
https://gitee.com/chuanjiao10/powershell/tree/master/sql/sqlserver

no timeout set on my script, 
default timeout is 15 second.  i tesetd, both on ps 5.1 and ps 7.4.

**Reply by IanElsinga** [reply]
Thanks @kasini3000, yes I am using code like that. The problem is that when a Connection Timeout is included that is less than 15s (10s in your example), and the instance is offline, it waits the for 15s in powershell 7.2 and 7.3.4 (didn't try 7.4) before returning with an error instead of returning after 10s. In PowerShell 5.1 it honors the timeout, and returns after 10 seconds as it should.
